### PR TITLE
Add protocol buffers to ballerina tool guide to tools-ides section

### PR DESCRIPTION
**Pull request by daneshk**
## Purpose
This PR is to add protocol buffers to ballerina tool guide to tools-ides section

## Related PRs
No

- Any checklist items to be verified before production deployment? no
- Special styling concerns? no
- Related media files? no

### CODE REVIEW

File: website/two-column-pages/docs/learn/tools-ides.md [hunk 1]
```diff
@@ -20,6 +21,10 @@ The graphical visualization tool is embedded in the Visual Studio Code plug-in.
 
 You can use the existing OpenAPI/Swagger files to generate connectors and services in Ballerina code. For details, see the [OpenAPI to Ballerina Code Generator](https://ballerina.io/learn/by-guide/open-api-based-service/).
 
+## Protocol Buffers to Ballerina code generator
+
+You can use the existing Protocol Buffers definition to generate connectors and services in Ballerina code. For details, see the [Protocol Buffers to Ballerina Code Generator](tools-ides/proto-ballerina-tool)
+
 ## API documentation generator
 
 As you develop new connectors, remote functions in connectors and other functions that you want to share with others, it's important to add API documentation that describes each entity and how it's used. Ballerina provides a framework called **Docerina** that generates API documentation from the annotations in your Ballerina files. You can check it out [here](https://github.com/ballerina-platform/ballerina-lang/tree/master/misc/docerina).
```
Comment (praneesha) [inline website/two-column-pages/docs/learn/tools-ides.md]:
```suggestion
As you develop new connectors, it's important to add API documentation that describes each entity (remote functions in connectors and other functions that you want to share with others) and how they are used.

Ballerina provides a framework called [Docerina](https://github.com/ballerina-platform/ballerina-lang/tree/master/misc/docerina), which generates API documentation from the annotations in your Ballerina files. 
```

File: website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# Protocol Buffers to Ballerina Tool
+
+The Protocol Buffers to Ballerina tool provides capability to generate Ballerina source code for the Protocol
```
Comment (praneesha) [inline website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md]:
```suggestion
The Protocol Buffers of the Ballerina tool provides capabilities to generate Ballerina source code using the Protocol
```

File: website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ Buffers definition. The `grpc` command in Ballerina is used to Protocol Buffers to Ballerina generation. Code
```
Comment (praneesha) [inline website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md]:
```suggestion
 Buffers definition. The `grpc` command in Ballerina is used for Protocol Buffers in Ballerina generation. The code
```

File: website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+ generation tool can produce `ballerina stub files` and `ballerina service/client template files`.
```
Comment (praneesha) [inline website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md]:
```suggestion
 generation tool can produce `ballerina stub` and `ballerina service/client template` files.
```

File: website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+ 
+> In Ballerina, Protocol Buffers serialization only supports in gRPC module. So currently you can use this tool to
```
Comment (praneesha) [inline website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md]:
```suggestion
> In Ballerina, Protocol Buffers serialization is only supported in the gRPC module. Therefore, currently, you can use this tool to
```

File: website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+You can generate Ballerina source code using the following command:
+
+```
+./ballerina grpc --input <proto-file-path> [--output <path>] [--mode client | server]
+```
+
+____**Options**____
+
+`--input`  - Path of the input .proto file. This is mandatory field. You need to provide the path of the definition
```
Comment (praneesha) [inline website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md]:
```suggestion
`--input`  - Path of the input .proto file. This is a mandatory field. You need to provide the path of the definition
```

File: website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+ file.
+
+`--output` - Location of the generated Ballerina source files. This is an optional field. 
+If output path is not specified, output will be written to a directory corresponding to the package in the Protocol
+ Buffers definition. 
+If package is not specified, output will be written to a 'temp' directory in the current location.
+
+`--mode`   - Set mode (client or server) to generate code samples. If not specified, only the stub file is generated.
```
Comment (praneesha) [inline website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md]:
```suggestion
`--mode`   - Set the mode (client or server) to generate code samples. If not specified, only the stub file is generated.
```

File: website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+
+## Sample
+
+Let's say you need to generate Ballerina source code to the following Protocol Buffers definition(in helloworld_service.proto file).
```
Comment (praneesha) [inline website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md]:
```suggestion
The below example shows how you can generate Ballerina source code from the following Protocol Buffers definition (in the `helloworld_service.proto` file).
```

File: website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+message HelloRequest {
+	string name = 1;
+}
+message HelloResponse {
+	string message = 1;
+}
+```
+
+* You can run the following command to generate client/service stub and service template.
```
Comment (praneesha) [inline website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md]:
```suggestion
* You can run the following command to generate the client/service stub and service template.
```

File: website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+```
+$ ballerina grpc --input helloworld_service.proto --mode service --output service
+```
+Once you execute the command, stub file(`helloworld_service_pb.bal`) and service template file(`helloWorld_sample_service.bal`) are generated inside service directory.
```
Comment (praneesha) [inline website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md]:
```suggestion
Once you execute the command, the stub file(`helloworld_service_pb.bal`) and the service template file (`helloWorld_sample_service.bal`) are generated inside the service directory.
```

File: website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+
+* You can run the following command to generate client/service stub and client template.
```
Comment (praneesha) [inline website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md]:
```suggestion
* You can run the following command to generate the client/service stub and client template.
```

File: website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+```
+$ ballerina grpc --input helloworld_service.proto --mode client --output client
+```
+Once you execute the command, stub file(`helloworld_service_pb.bal`) and service template file(`helloWorld_sample_client.bal`) are generated inside client directory.
```
Comment (praneesha) [inline website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md]:
```suggestion
Once you execute the command, the stub file(`helloworld_service_pb.bal`) and the service template file (`helloWorld_sample_client.bal`) are generated inside the client directory.
```

File: website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+
+* You can run the following command to generate only client/service stub.
```
Comment (praneesha) [inline website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md]:
```suggestion
* You can run the following command to generate only the client/service stub.
```

File: website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+```
+$ ballerina grpc --input helloworld_service.proto --output stubs
+```
+Once you execute the command, only stub file(`helloworld_service_pb.bal`) is generated inside stubs directory.
```
Comment (praneesha) [inline website/two-column-pages/docs/learn/tools-ides/proto-ballerina-tool.md]:
```suggestion
Once you execute the command, only the stub file (`helloworld_service_pb.bal`) is generated inside the `stubs` directory.
```

File: website/two-column-pages/docs/learn/tools-ides.md [hunk 15]
```diff
@@ -4,6 +4,7 @@ Ballerina provides language servers, editors, IDEs, and graphical visualization
 
 - [Editor and IDE support](#editor-and-ide-support)
 - [OpenAPI to Ballerina code generator](#openapi-to-ballerina-code-generator)
+- [Protocol Buffers to Ballerina code generator](#protocol-buffers-to-ballerina-code-generator)
```
[the remark this answers is not in the document]
Comment (daneshk) [reply website/two-column-pages/docs/learn/tools-ides.md]: This needs to be "Protocol Buffers to Ballerina". same like "OpenAPI to Ballerina"

**Pull request merged**